### PR TITLE
std.datetime - start using ranges

### PR DESCRIPTION
**Pull request by WalterBright**
Also removed pointless code that checked if Linux system APIs threw D non-recoverable exceptions. This is not a practice that we should endorse.

### CODE REVIEW

**Comment by schveiguy** [comment]
Looks like these range changes increase the code size.

**Comment by WalterBright** [comment]
Yeah, they will, because the inlining of all those ranges adds code.

**Comment by WalterBright** [comment]
https://github.com/D-Programming-Language/dmd/pull/4754

File: std/datetime.d [hunk 1]
```diff
@@ -28075,13 +28075,15 @@ assert(tz.dstName == "PDT");
...
+        import std.path : toNormalizedPath, chainPath;
+        import std.conv : to;
 
         name = strip(name);
 
         enforce(tzDatabaseDir.exists(), new DateTimeException(format("Directory %s does not exist.", tzDatabaseDir)));
         enforce(tzDatabaseDir.isDir, new DateTimeException(format("%s is not a directory.", tzDatabaseDir)));
 
-        immutable file = buildNormalizedPath(tzDatabaseDir, name);
+        const file = toNormalizedPath(chainPath(tzDatabaseDir, name)).to!string;
```
Comment (andralex) [inline std/datetime.d]: y u no immutable

**Comment by andralex** [comment]
alrite

**Comment by andralex** [comment]
Auto-merge toggled on

**Pull request merged**